internal/encryption/keyprovider/static/config.go

```diff
@@ -0,0 +1,19 @@
+package static
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/opentofu/opentofu/internal/encryption/keyprovider"
+)
+
+type Config struct {
+	Key string `hcl:"key"`
```

Comment:
This will fail the example in the design doc you provided with pbkdf2.

In practice, as mentioned in my design document comment I would suggest both:
1. Using hex as the field name for passing a key in hex format.
2. Passing the passphrase to pbkdf2 directly. I don't think having a shared "resource" for both static keys and passphrases is a good idea.